src/bin/pg_autoctl/env_utils.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * src/bin/pg_autoctl/env_utils.c
+ *   Utility functions for interacting with environment settings.
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the PostgreSQL License.
+ *
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "postgres_fe.h"
+
+#include "env_utils.h"
+#include "log.h"
+
+/*
+ * get_env_variable checks for the environment variable and copies
+ * its value into provided buffer if the buffer is not null.
+ *
+ * Function returns the length of the environment variable.
+ *
+ * Return values
+ * - (-1) if variable is not found
```

Comment:
I think it makes sense to add another return code here:
-2 Error while getting environment variable

That way this function can actually check if the environment variable fits into the buffer, and fail otherwise. It's always nice to take away responsibility from the caller as much as possible. (-2 can also be returned in the error check at the start of the function)

I think it's also useful for readability to give these values a name using `#define`. e.g.:
```c
#define ENV_ERROR -2
#define ENV_NOT_FOUND -1
#define ENV_EMPTY 0
```

Reply:
If we need to do that, then we should return a boolean and handle the variable content and size separately, and then maybe a copy of the environment variable name.

``` c
typedef struct EnvVar
{
    char name[BUFSIZE];
    char value[BUFSIZE];
    size_t valueSize;
} EnvVar;

...

    EnvVar pg_regress_sock_dir = { "PG_REGRESS_SOCK_DIR", { 0 }, 0 };
    
    if (!getEnvVar(EnvVar))
    {
         /* errors have already been logged */
        return false;
    }    
```

Reply:
added but not sure if it is helpful.  anything < 0 is error case.